### ISSUE TITLE
Support for ssh-agent and forwarding?

### ISSUE THREAD

**Issue by rupor-github** [reconquest/orgalorg#12]
Hi, are there any plans to support this?

**Comment by seletskiy** [comment]
@rupor-github: Hi! Not yet. It was already mentioned in the #10. I'l glad to merge PR with ssh-agent support.

**Comment by rupor-github** [comment]
Here you are: https://github.com/reconquest/orgalorg/pull/14

**Comment by rupor-github** [comment]
@seletskiy did you have a chance to look at my pool requests (both runcmd and orgalorg)?

**Comment by seletskiy** [comment]
@rupor-github: Sorry, I was little bit of out context lately. I will take a close look soon. Thanks for poking me.

**Comment by linuxbsdfreak** [comment]
@seletskiy  - Do you know if this can be merged? I am using the orgalorg tool and would like to use the ssh-agent withougt entering the passphrase evertime.

**Comment by prologic** [comment]
Looks like the PR never got merged. What happening here?

**Comment by seletskiy** [comment]
@prologic it seems that `runcmd` library repo was lost along with required PR, so author closed PR.

We need to restore this feature in `runcmd` first.

**Comment by prologic** [comment]
This was some years ago... Is there still interest in adding this feature? I'm not seeing much evidence of activity on this project of late... Do you still actively maintain/develop it?

**Comment by seletskiy** [comment]
@prologic: well, I do not use `orgalorg` right now, but that doesn't mean, that project is dead.
If someone will add agent functionality back in `runcmd` then I will gladly merge PR #14.

**Comment by kovetskiy** [comment]
@prologic yes, I use the program on an everyday basis, the thing about it is that it just works fine.

**Comment by tbellembois** [comment]
I undig a very old topic.

Is there any possibility to use the `https://godoc.org/golang.org/x/crypto/ssh` library instead of `runcmd` ?
`runcmd` does not look to be maintained.

**Comment by kovetskiy** [comment]
@tbellembois what point do you see in it? The runcmd package doesn't do anything except wrapping golang.org/x/crypto/ssh, providing an interface which we use in orgalorg itself as abstraction layer for Local/Remote runners.

**Comment by kovetskiy** [comment]
@tbellembois nevermind, I've already forked `runcmd` and reworked it a bit. 

Also, I've added support of ssh-agent here: https://github.com/reconquest/orgalorg/commit/ed1e994d72bbd39ea729c21bbd6cbb7294ab6c36

**Comment by seletskiy** [comment]
![flash](https://media1.giphy.com/media/lRnUWhmllPI9a/giphy.gif)
@kovetskiy

**Comment by tbellembois** [comment]
Orgalorg keeps asking for my ssh keys.
Changing the `main.go` file into (line 707):
```go
	if os.Getenv("SSH_AUTH_SOCK") != "" {
		sock, err := net.Dial("unix", os.Getenv("SSH_AUTH_SOCK"))
		if err != nil {
			return nil, hierr.Errorf(
				err,
				"unable to dial to ssh agent socket: %s",
				os.Getenv("SSH_AUTH_SOCK"),
			)
		}

		keyring = agent.NewClient(sock)
	} else {
		keyring = agent.NewKeyring()
	}
```
does the job but not sure this is the right think to do...

**Comment by seletskiy** [comment]
@tbellembois feel free to open pull request!

**Comment by tbellembois** [comment]
Done ! Thanks.
